src/tracks/MidiClip.cpp

```diff
@@ -344,6 +344,49 @@ void MidiClip::splitNotes(const NoteVector& notes, TimePos pos)
 	}
 }
 
+void MidiClip::splitNotesAlongLine(TimePos pos1, int key1, TimePos pos2, int key2, bool deleteShortEnds)
+{
+	if (m_notes.empty()) { return; }
+
+	// Don't split if the line is horitzontal
+	if (key1 == key2) { return; }
+
+	addJournalCheckPoint();
+
+	const auto notesCopy = m_notes;
+	const auto slope = 1.f * (pos2 - pos1) / (key2 - key1);
+	const auto& [minKey, maxKey] = std::minmax(key1, key2);
+
+	for (const auto& note : notesCopy)
+	{
+		// Continuing if the key is <= to minKey, since the line is drawn from the top of minKey to the top of maxKey, but only passes through maxKey - minKey - 1 total keys.
```

Comment:
I like the term "skip" better.

```suggestion
		// Skip if the key is <= to minKey, since the line is drawn from the top of minKey to the top of maxKey, but only passes through maxKey - minKey - 1 total keys.
```